### PR TITLE
Add-ngrok-and-terminal-improvements

### PR DESCRIPTION
**Pull request by victoriaplummer**

### CODE REVIEW

File: Data Client/server.js [hunk 1]
```diff
@@ -106,8 +119,49 @@ app.get("/sites", jwt.authenticateSessionToken, async (req, res) => {
   }
 });
 
-// Start the server
-const PORT = process.env.PORT || 3000;
-app.listen(PORT, () => {
-  console.log(`Server is running on http://localhost:${PORT}`);
-});
+// Start server with NGROK
+const startServer = async () => {
```
Comment (zplata) [inline Data Client/server.js:136]: Can we make this a standalone script mapped to its own command in package.json (i.e. `npm run ngrok`) ? It'll be easier for local dev in general to just run the server start and ngrok start on two different instances

File: Data Client/utils/ngrok-url.txt [hunk 2]
```diff
@@ -0,0 +1 @@
+https://2c90-66-65-130-58.ngrok-free.app
```
Comment (zplata) [inline Data Client/utils/ngrok-url.txt]: Is this file required?

File: Designer Extension/src/main.js [hunk 3]
```diff
@@ -79,7 +70,23 @@ const App = () => {
     } else {
       exchangeAndVerifyIdToken();
     }
-  }, []);
+
+    // Listen for message from the OAuth callback window
+    const handleAuthComplete = (event) => {
+      if (
+        // event.origin === "http://localhost:3000" &&
+        event.data === "authComplete"
```
Comment (zplata) [inline Designer Extension/src/main.js:81]: Nice 🙌

File: Data Client/server.js (hunk 1 above)
Comment (victoriaplummer) [reply Data Client/server.js:136]: Would we still be able to serve the ngrok link in the terminal so they can easily copy/paste? I wouldn't want to lose this functionality

Comment (victoriaplummer) [reply Data Client/server.js:136]: Canva is able to do it - [I'm gonna dig around here and see what we can emulate](https://github.com/canva-sdks/canva-apps-sdk-starter-kit/blob/b49bf55679121374c2a146e8e63a613754f372d1/scripts/start/app_runner.ts)

Comment (zplata) [reply Data Client/server.js:136]: I don't think it needs to be tied to the express server start here, it could be its own node standalone file we run (and can still display the table in the terminal for copy paste). But if we are somehow able to retain the ngrok context on server restarts, all for it! Haven't tried out Canva's example
